src/initialize_dae.jl

```diff
@@ -136,13 +136,14 @@ function _initialize_dae!(integrator, prob::ODEProblem, alg::ShampineCollocation
     nlequation! = function (out,u)
       update_coefficients!(M,u,p,t)
       #M * (u-u0)/dt - f(u,p,t)
-      @. tmp = (u - u0)/dt
+      tmp = @. (u - u0)/dt
```

Comment:
why?

Reply:
oh, the dual numbers. Can we make a DualCache here?

Reply:
I think DualCache is not compatible with arbitrary array length.